libsolidity/formal/BMC.cpp

```diff
@@ -162,11 +162,24 @@ void BMC::endVisit(FunctionDefinition const& _function)
 		smtutil::Expression constraints = m_context.assertions();
 		checkVerificationTargets(constraints);
 		m_verificationTargets.clear();
+		m_pathConditions.clear();
 	}
 
 	SMTEncoder::endVisit(_function);
 }
 
+pair<SMTEncoder::VariableIndices, smtutil::Expression> BMC::visitBranch(ASTNode const& _statement, smtutil::Expression const& _condition)
+{
+	auto indicesBeforeBranch = copyVariableIndices();
+	pushPathCondition(_condition);
+	_statement.accept(*this);
+	auto pathConditionOnExit = currentPathConditions();
+	popPathCondition();
+	auto indicesAfterBranch = copyVariableIndices();
+	resetVariableIndices(indicesBeforeBranch);
+	return {indicesAfterBranch, pathConditionOnExit};
+}
```

Comment:
This is almost identical to `SMTEncoder::visitBranch`. Why not make this the single function in SMTEncoder? Callers that don't care about the `pathConditionsOnExit` can just ignore it.

Reply:
That's a good point. For some reason I am becoming more and more convinced that the path condition is needed only in BMC, so I guess I wanted to keep the modified version of `visitBranch` there. But I think your suggestion is good, it avoids code duplication.

Reply:
I agree that the path condition is needed by BMC only, but the function is almost the same, so I think it's fine to move it to SMTEncoder and just ignore the path conditions when not BMC.